scripts/covid_hosp_prod.R

```diff
@@ -97,226 +111,266 @@ rlang::list2(
           mutate(
-            forecaster = names(forecaster_fns[forecasters]),
-            geo_value = as.factor(geo_value)
+            geo_value = ifelse(geo_value == "usa", "us", geo_value),
+            time_value = time_value - 3
           )
-      },
-      pattern = cross(aheads, forecasters)
-    ),
-    tar_target(
-      name = ensemble_res,
-      command = {
-        forecast_res %>%
-          ensemble_linear_climate(
-            aheads,
-            other_weights = geo_forecasters_weights,
-            max_climate_ahead_weight = 0.6,
-            max_climate_quantile_weight = 0.6
-          ) %>%
-          filter(geo_value %nin% geo_exclusions) %>%
-          ungroup() %>%
-          sort_by_quantile()
-      },
-    ),
-    tar_target(
-      name = ensemble_mixture_res,
-      command = {
-        forecast_res %>%
-          ensemble_linear_climate(
-            aheads,
-            other_weights = geo_forecasters_weights,
-            max_climate_ahead_weight = 0.6,
-            max_climate_quantile_weight = 0.6
-          ) %>%
-          filter(geo_value %nin% geo_exclusions) %>%
-          ungroup() %>%
-          bind_rows(forecast_res %>%
-            filter(forecaster %in% c("windowed_seasonal_extra_sources")) %>%
-            filter(forecast_date < target_end_date)) %>% # don't use for neg aheads
-          group_by(geo_value, forecast_date, target_end_date, quantile) %>%
-          summarize(value = mean(value, na.rm = TRUE), .groups = "drop") %>%
-          sort_by_quantile()
-      },
-    ),
-    tar_target(
-      name = forecasts_and_ensembles,
-      command = {
-        bind_rows(
-          forecast_res,
-          ensemble_res %>% mutate(forecaster = "ensemble"),
-          ensemble_mixture_res %>% mutate(forecaster = "ensemble_mix"),
-          # TODO: Maybe later, match with flu_hosp_prod
-          # ensemble_mixture_res_2 %>% mutate(forecaster = "ensemble_mix_2"),
-          # combo_ensemble_mixture_res %>% mutate(forecaster = "combo_ensemble_mix")
+      } else {
+        train_data <-
+          nhsn_latest_data %>%
+          data_substitutions(disease = "covid", last(forecast_generation_dates)) %>%
+          as_epi_df(as_of = as.Date(forecast_date_int)) %>%
+          mutate(time_value = time_value - 3)
+      }
+      nssp <- current_nssp_archive %>%
+        epix_as_of(min(as.Date(forecast_date_int), current_nssp_archive$versions_end)) %>%
+        mutate(time_value = time_value)
+      attributes(train_data)$metadata$as_of <- as.Date(forecast_date_int)
+      train_data %>%
+        forecaster_fns[[forecaster_fn_names]](ahead = aheads, extra_data = nssp) %>%
+        mutate(
+          forecaster = forecaster_fn_names,
+          geo_value = as.factor(geo_value)
         )
+    },
+    pattern = map(aheads)
+  )
+)
+combined_forecasts <- tar_combine(
+  name = forecast_full,
+  forecast_targets[["forecast_res"]],
+  command = {
+    dplyr::bind_rows(!!!.x)
+  }
+)
```

Comment:
So an unfortunate thing this will result in is if any of the forecasts change for any forecast_date, every target downstream from this is going to get invalidated. Not sure how long scoring takes, but this could get annoying

Reply:
less sure about an immediate alternative.

Reply:
hm yea. i did it this way because i think i needed it to resolve the "forecasts always rerunning" issue. im not 100% on it, but this is the way it's setup in the explore pipeline and that pipeline doesnt have spurious cache invalidation issues.

Reply:
oh I forgot we actually did it this way in explore. My memory was that that did invalidate everything downstream of the single target, but it's been a while since I've run it

Reply:
no i agree, i think things downstream of this will change if any component of this changes. but the choice is between "everything reruns all the time" and "things downstream of this aggregate always rerun", which seems like an upgrade.